### PR TITLE
Added section on adversarial bias mitigation to fairness guide

### PR DESCRIPTION
**Pull request by ArjunSubramonian**

### CODE REVIEW

File: chapters/part3/fairness.md [hunk 1]
```diff
@@ -722,9 +722,289 @@ We have made the [bias-mitigated, finetuned large RoBERTA model for SNLI](https:
...
+
+![adversarial_equation_1](/part3/fairness/adversarial_equation_1.png)
+
+We achieve this non-traditional parameter update scheme in AllenNLP using an [`on_backward`](http://docs.allennlp.org/main/api/training/callbacks/callback/#on_backward) trainer callback. An [`on_backward`](http://docs.allennlp.org/main/api/training/callbacks/backward/) trainer callback performs custom backpropagation and allows for gradient manipulation; `on_backward` callback(s) will be executed in lieu of the default `loss.backward()` called by `GradientDescentTrainer`.
+
+We provide the [`AdversarialBiasMitigatorBackwardCallback`](http://docs.allennlp.org/main/api/fairness/adversarial_bias_mitigator/#adversarialbiasmitigatorbackwardcallback) to perform the non-traditional parameter updates required by adversarial bias mitigation.
+
+```python
+@TrainerCallback.register("adversarial_bias_mitigator_backward")
```
Comment (AkshitaB) [inline chapters/part3/fairness.md:759]: Perhaps it makes sense to just have the snippet with the actual parameter update in the guide; we already provide a link to the complete code.

File: chapters/part3/fairness.md [hunk 2]
```diff
@@ -722,9 +722,289 @@ We have made the [bias-mitigated, finetuned large RoBERTA model for SNLI](https:
...
+
+## Pitfalls of Adversarial Bias Mitigation
+
+Adversarial bias mitigation does have some pitfalls. For one, both the predictor and adversary are black boxes, which makes it uninterpretable in contrast to the bias mitigators described in previous sections. Furthermore, while [1] derives theoretical guarantees for adversarial bias mitigation, the proof relies on the strong, and impractical, assumption that the loss functions of the predictor and adversary are convex with respect to their parameters. Finally, achieving convergence in adversarial networks is tricky in practice, especially for large, contextual language models; some tips for training adversarial networks include to [2]:
```
Comment (AkshitaB) [inline chapters/part3/fairness.md:990]:
```suggestion
Adversarial bias mitigation does have some pitfalls. For one, both the predictor and adversary are black boxes, which makes it uninterpretable in contrast to the bias mitigators described in previous sections. Furthermore, while [1] derives theoretical guarantees for adversarial bias mitigation, the proof relies on the strong, and impractical, assumption that the loss functions of the predictor and adversary are convex with respect to their parameters. Finally, achieving convergence in adversarial networks is tricky in practice, especially for large, contextual language models; **some tips for training adversarial networks** include to [2]:
```

Comment (AkshitaB) [reply chapters/part3/fairness.md:990]: Note: the suggestion is to **bold** the tips part; I don't think that's visible in github edits.

**Review by AkshitaB — CHANGES_REQUESTED** [review]

**Review by AkshitaB — APPROVED** [review]
